public/app/features/query/state/PanelQueryRunner.ts

```diff
@@ -451,6 +470,33 @@ export class PanelQueryRunner {
   getLastRequest(): DataQueryRequest | undefined {
     return this.lastRequest;
   }
+
+  shouldAddErrorWhenDatasourceVariableIsMultiple(
+    datasource: DataSourceRef | DataSourceApi | null,
+    scopedVars: ScopedVars | undefined
+  ): boolean {
+    let addWarningMessageMultipleDatasourceVariable = false;
+
+    //If datasource is a variable
+    if (datasource?.uid?.startsWith('${')) {
+      const variableName = this.templateSrv.getVariableName(datasource.uid) ?? '';
+      // we can access the raw datasource variable values inside the replace function if we pass a custom format function
+      this.templateSrv.replace(datasource.uid, scopedVars, (value: string | string[]) => {
+        // if the variable has multiple values and it's does not have a scoped variable (meaning not being repeated by the variable)
+        if (
+          Array.isArray(value) &&
+          value.length > 1 &&
+          (scopedVars === undefined || (scopedVars && !scopedVars[variableName]))
```

Comment:
you can remove this condition

Reply:
🤔mmm the condition checks if the panel is being repeated by the same datasource variable. If the condition fails it means we need to add an error, otherwise, it meets the criteria, no?

Reply:
@axelavargas no your not correct (at least not in scenes where we do not use scopedVars for repeats). Because your using templateSrv.replace here the value you get here already has taken any scopedVars into account. So if there was a scopedVars here you would not get an array here. So you just have to check if the value is an array, if it's an array then the value is not scoped (by repeat)

Reply:
Oh, I see, I will check if this works in dashboards that are rendered without scenes. Thank you, Torkel 😄

Reply:
@torkelo I applied the PR suggestions, thanks for the clarification with the `scopedVars` and `templateSrv.replace` function (I learned something new)  🙏🏾